### PR TITLE
test(e2e):eap modal

### PR DESCRIPTION
**Pull request by STew790**
## Description

Adding e2e test of EAP joining process

## Related Issue

Resolve

### CODE REVIEW

File: packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ({ electronApp, window, localDataDir } = await launchSuite());
+    rmDirRecursive(localDataDir);
+});
+
+testPlaywright.afterAll(() => {
+    electronApp.close();
+});
+
+/**
+ * Test case:
```
Comment (ondracja) [inline packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts:21]: Pls simplify the test case, no need to go into so much detail, we can read the exact steps from the test itself.

File: packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+ * 1. Navigate to `Settings/Application`
+ * 2. Scroll down to the `Experimental features` part
+ * 3. In the `Early access program` modal, click on `Join`
+ * 4. tick the input and click on `Join the program`
+ * 5. click on `Check for updates now` and either of these two should appear
+ * 6. a notification `No new updates available`
+ * 7. a modal window that allows user to do download a beta version of the Suite
+ */
+testPlaywright('Join the early access program', async page => {
+    // Action
```
Comment (ondracja) [inline packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts]: These denotions are also not really important anymore, just separate the test setup part/assert by a blank line.

File: packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    await onTopBar.openSettings(window);
+    await onSettingsPage.goToDesiredSettingsPlace(window, 'general');
+    await onSettingsPage.joinEarlyAccessProgram(window);
+
+    // Assertion
+
+    const buttonText = 'Leave';
```
Comment (ondracja) [inline packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts]: Const definitions should be at the top of the test.

**Comment by ondracja** [comment]
Also check the `type-check_and_lint` job, is your linter running?

File: packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts (hunk 1 above)
Comment (STew790) [reply packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts:21]: Test case simplified

File: packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts (hunk 2 above)
Comment (STew790) [reply packages/suite-desktop-core/e2e/tests/general/eap-modal.test.ts]: Deleted

**Review by ondracja — APPROVED** [review]

**Comment by ondracja** [comment]
Just FYI:  the test is not going to run with current setup (it would need to be added to the `test.yaml` test matrix) but it's ok for now as I'm in middle of migrating the flow to github actions and will include it later.

**Pull request merged**